### PR TITLE
PLAT-127115: Fixed Input label when value is 0

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
- The issue was discovered in the qa-sampler for a picker with add/remove items which contains an input, with default value 0 Click on the index input field. --> Input '0'. --> Index Button's inner text is still 'Index'

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Updated the computed value for ButtonLabel inside InputBase to take a value of `0` as a valid value

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-127115

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

**Review by adrian-cocoara-lgp — APPROVED** [review]
LGTM

File: Input/Input.js [hunk 1]
```diff
@@ -474,7 +474,7 @@ const InputBase = kind({
 			return calcAriaLabel('', null, placeholder);
 		},
 		buttonLabel: ({placeholder, type, value}) => {
-			return (isPasswordType(type) ? convertToPasswordFormat(value) : value) || placeholder;
+			return (isPasswordType(type) ? convertToPasswordFormat(value) : value) || typeof value === 'number' ? value.toString() : placeholder;
```
Comment (0x64) [inline Input/Input.js:477]: To make this logic more readable, what about updating this code like the below pseudo code?

```
if (value's type is not one of String or Number) {
	return placeholder;
} else {
	return isPasswordType(type) ? convertToPasswordFormat(value) : value
}
```

Comment (0x64) [reply Input/Input.js:477]: Also, I found the same issue of value 0 during review from the upper function `buttonAriaLabel`.

```
		buttonAriaLabel: ({placeholder, type, value}) => {
			if (value) { // This line is to check whether value is null or undefined due to a missing prop value. So, type checking looks better here.
				...
			}
			...
		},
```

File: Input/Input.js [hunk 2]
```diff
@@ -466,15 +466,19 @@ const InputBase = kind({
...
 				type = isPasswordType(type) ? 'password' : type;
 				return calcAriaLabel('', type, type === 'number' ? value.split('') : value);
 			}
 
 			return calcAriaLabel('', null, placeholder);
 		},
 		buttonLabel: ({placeholder, type, value}) => {
-			return (isPasswordType(type) ? convertToPasswordFormat(value) : value) || placeholder;
+			if (value || value === 0) {
+				return isPasswordType(type) ? convertToPasswordFormat(value) : value.toString();
```
Comment (0x64) [inline Input/Input.js:478]: It looks that we don't need `.toString()` anymore.

Comment (0x64) [reply Input/Input.js:478]: Oh, I see this is required. 👍

**Pull request merged**